include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8793,4 +8793,32 @@ def err_typecheck_non_count_return_bounds : Error<
 def err_typecheck_bounds_type_annotation_identifier: Error<
   "type name cannot have identifier in it">;
 
+def err_typecheck_bounds_type_annotation_for_illegal_type : Error<
+  "interface type only allowed for a declaration with pointer "
```

Comment:
I have the same concern as you do.   The clang developer documentation urges you to keep error messages under 80 characters and to not repeat information or provide unnecessary information.    I'm trying to stay in-line with that.  In this case, the error message will point directly at the itype declaration.  I hope this will give the user the context needed to understand this.

Here an example of an error message:
```
interop_type_annotations.c:78:18: error: interface type only allowed for a declaration with pointer or array type
void f1a(int p : itype(ptr<int>)) {   // expected-error {{interface type only allowed for a declaration with pointer or array type}}
                 ^
```
I could change the error message to use `bounds-safe interface type` instead of just `interface type`.  What are your thoughts?